### PR TITLE
Improve api reference docs

### PR DESCRIPTION
**Pull request by harveyxia**
BOT NOTES: 
resolves https://github.com/solo-io/gloo-mesh/issues/1153

### CODE REVIEW

**Comment by harveyxia** [comment]
To see the new reference docs, `cd docs && make serve-site`

File: docs/docsgen/generate_proto_docs.go [hunk 1]
```diff
@@ -0,0 +1,255 @@
...
+			return filepath.Base(str)
+		},
+		"link_to_type": func(v interface{}) string {
+			switch fieldType := v.(type) {
+			case *gendoc.MessageField:
+				link, ok := links[fieldType.FullType]
+				if ok {
+					return link
+				} else if strings.Contains(fieldType.FullType, ".") {
+					fmt.Printf("link not found for %s", fieldType)
```
Comment (jenshu) [inline docs/docsgen/generate_proto_docs.go:249]: just curious, why use both fmt.Printf and log.Printf in this file? do they behave differently?

File: docs/content/reference/api/_index.md [hunk 2]
```diff
@@ -1,12 +1,77 @@
...
+
+  - [TrafficTarget]({{< ref "traffic_target.md#discovery.mesh.gloo.solo.io.TrafficTargetSpec" >}})
+
+  - [Workload]({{< ref "workload.md#discovery.mesh.gloo.solo.io.WorkloadSpec" >}})
+
+
+
+### extensions.networking.mesh.gloo.solo.io
+
+  - [extensions.networking.mesh.gloo.solo.io.NetworkingExtensions]({{< ref "networking_extensions.md#extensions.networking.mesh.gloo.solo.io.NetworkingExtensions" >}})
```
Comment (jenshu) [inline docs/content/reference/api/_index.md:35]:
```suggestion
  - [NetworkingExtensions]({{< ref "networking_extensions.md#extensions.networking.mesh.gloo.solo.io.NetworkingExtensions" >}})
```

Comment (harveyxia) [reply docs/content/reference/api/_index.md:35]: I think it makes sense to include the qualifying package

File: docs/docsgen/generate_proto_docs.go (hunk 1 above)
Comment (harveyxia) [reply docs/docsgen/generate_proto_docs.go]: among other things, the `log` package provides a `Fatal` convenience method. I'll switch to outputting with `log` to be consistent

**Review by joekelley — APPROVED** [review]

**Pull request merged**